src/main/java/pl/psnc/dei/ui/components/batches/ComplexBatchImportComponent.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    private void initUploads() {
+        this.prepareNameTextFiled();
+        this.prepareFileSelectButton();
+        this.prepareImportFile();
+        this.prepareImportButton();
+    }
+
+    private void prepareFileSelectButton() {
+        this.fileSelectButton = new Button();
+        this.fileSelectButton.setText("Upload");
+        this.fileSelectButton.setIcon(new Icon(VaadinIcon.CLOUD_UPLOAD_O));
+    }
+
+    private void prepareImportFile() {
+        this.memoryBuffer = new MemoryBuffer();
+        this.file = new Upload(this.memoryBuffer);
+        this.file.setMaxFiles(1);
+        this.file.setAcceptedFileTypes(".csv");
+        this.file.setDropAllowed(true);
+        this.file.setUploadButton(this.fileSelectButton);
+        this.file.addFileRejectedListener(
+                event -> Notification.show("File rejected. Reason: " + event.getErrorMessage(), 3000, Notification.Position.MIDDLE)
+        );
+        this.file.getElement().addEventListener("upload-abort", new DomEventListener() {
+            @Override
+            public void handleEvent(DomEvent domEvent) {
+                memoryBuffer = new MemoryBuffer();
+                file.setReceiver(memoryBuffer);
+            }
+        });
+        this.file.setDropLabel(new Label("Upload up to one file to make import from"));
+        this.file.addClassName("flex-1");
+        this.file.addClassName("margin-top-16px");
+    }
+
+    private void prepareNameTextFiled() {
+        this.nameTextFiled = new TextField();
+        this.nameTextFiled.setLabel("Import name");
+        this.nameTextFiled.addClassName("flex-1");
+    }
+
+    private void prepareImportButton() {
+        this.importButton = new Button();
+        this.importButton.setText("Import");
+        this.importButton.addClickListener(
+                event -> this.sendImport());
+    }
+
+    private Boolean validate() throws IOException {
+        if (this.aggregatorSelect.getValue() == null){
+            Notification.show("Select Aggregator", 3000, Notification.Position.MIDDLE);
+            return false;
+        } else if (this.projectSelect.getValue() == null) {
+            Notification.show("Select Project", 3000, Notification.Position.MIDDLE);
+            return false;
+        } else if (this.memoryBuffer.getInputStream().readAllBytes().length == 0) {
+            Notification.show("Select File", 3000, Notification.Position.MIDDLE);
+            return false;
+        }
+        return true;
+    }
+
+    private void prepareImportName() {
+        if (this.nameTextFiled.getValue().isBlank()) {
+            this.nameTextFiled.setValue(ImportNameCreatorUtil.generateImportName(this.projectSelect.getValue().getName()));
+        } else {
+            this.nameTextFiled.setValue(
+                    this.nameTextFiled.getValue().strip()
+            );
+        }
+    }
+
+    private void sendImport() {
+        try {
+            this.prepareImportName();
```

Comment:
I don't think it is good idea to generate import name and put it into the field just to read it below. You could just generate the import name and keep it in string, at most displaying the name in notification.
Moreover, you put this instruction in very inappropriate place, because it is before validations. In the situation, when e.g. no file is uploaded, import name with current timestamp (with an accuracy of seconds) will be generated. Then after uploading correct file, you still have this import name, but with outdated timestamp.

Reply:
Populating field has purpose. So far we do not redirecting user after sending import, thus field should have some value.
I am not sure if auto generated name here has purpose of timestamp, as `import` has `creationDate` field anyway. I consider import name just as name which could be whatever we want and I am not sure if expected behavior is to change it every time user click button

Reply:
This code will be refactored to incorporate `Project` name change